test/Microsoft.Framework.PackageManager.FunctionalTests/DnuCommandsTests.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Framework.CommonTestUtils;
+using Microsoft.Framework.Runtime.DependencyManagement;
+using Xunit;
+
+namespace Microsoft.Framework.PackageManager
+{
+    public class DnuCommandsTests
+    {
+        public static IEnumerable<object[]> RuntimeComponents
+        {
+            get
+            {
+                return TestUtils.GetRuntimeComponentsCombinations();
+            }
+        }
+
+        private void InstallFakeApp(string rootDir, string name, string version)
+        {
+            Directory.CreateDirectory($"{rootDir}/packages/{name}/{version}/app");
+            File.WriteAllText($"{rootDir}/packages/{name}/{version}/app/{name}.cmd", "");
+            File.WriteAllText($"{rootDir}/{name}.cmd", $"~dp0/packages/{name}/{version}/app/{name}.cmd".Replace('/', Path.DirectorySeparatorChar));
+            File.WriteAllText($"{rootDir}/packages/{name}/{version}/{name}.{version}.nupkg.sha512", "TestSha");
+        }
+
+        private void WriteLockFile(string dir, string libName, string version)
+        {
+            var lockFile = new LockFile
+            {
+                Islocked = false,
+                Libraries = new List<LockFileLibrary>
+                                {
+                                    new LockFileLibrary
+                                    {
+                                        Name = libName,
+                                        Version = new NuGet.SemanticVersion(version),
+                                        Sha = "TestSha"
+                                    }
+                                }
+            };
+            var lockFormat = new LockFileFormat();
+            lockFormat.Write($"{dir}/project.lock.json", lockFile);
+        }
+
+        [Theory]
+        [MemberData("RuntimeComponents")]
+        public void DnuCommands_Uninstall_PreservesPackagesUsedByOtherInstalledApps(string flavor, string os, string architecture)
```

Comment:
Add 2 more test that does:
First:
1. Add a package that is not used by anyone
2. Have a single application installed (that doesn't use that packge)
3. Uninstall the app
4. Make sure there is no package left

Second:
Exactly like the previous test but pass `--no-purge` and make sure that the orphan package is not deleted